migrating_from_ocp_3_to_4/planning-migration-3-4.adoc

```diff
@@ -73,6 +73,82 @@ For more information, see xref:../updating/updating-cluster-between-minor.adoc#u
+
+* Internal services provided by the cluster, such as a web console, open authentication, alert manager, and metrics, use the default DNS domain in their URLs. With the exception of the web console, these URLs cannot be easily changed.
+
+If preserving the DNS domain of migrated applications is a requirement, several options exist.
+
+==== Option 1: Isolate the DNS domain of the target cluster from the clients
+
+To hide the {product-title} 4 DNS domain from the clients, a network element such as an application load balancer or a reverse proxy is placed between the clients and the {product-title} 4 cluster.
+
+The applications are migrated from the source to the target cluster and receive a FQDN in the default DNS domain of the target cluster (`app1.apps.ocp4.example.com`).
+
+The original application FQDN (`app1.apps.ocp3.example.com`) is updated in the DNS server to return the IP address of the network device. The network device contains rules to send requests received for the application in the source domain to the load balancer in the target cluster, using the target cluster domain.
+
+----
+app1.apps.ocp3.example.com ---|reverse proxy|--->  app1.apps.ocp4.example.com
+----
+
+The applications can be migrated using any of the strategies described in Network traffic migration strategies {xref TBD}.
+
+You can migrate one application at a time, creating a wildcard DNS record for the `\*.apps.ocp3.example.com` domain that points to the IP address of the load balancer of the source cluster. You then create a specific DNS record for each application, which points to the IP address of the network device in front of the target cluster. Because a specific DNS record has higher priority than a wildcard record, no conflict arises when the application FQDN is resolved.
+
+Some additional considerations for this option are:
+
+* The network device must terminate all secure TLS connections. Otherwise, if the connections are passed through to the OpenShift load balancer, the FQDN of the target application will be exposed to the client, and certificate errors will occur.
+
+* The applications must support this configuration, and they must not return links referencing the target cluster domain to the clients. If such links are leaked back to the client within the returned content, parts of the application may not load or work properly.
+
+==== Option 2: Set up the target cluster to accept the source DNS domain
+
+You can set up the target cluster to accept requests for migrated applications in the same DNS domain that was defined in the source cluster.
+
+* Non-secure (HTTP) access:
+
+  ** For the router in the default ingress controller of the target cluster to accept requests for applications in the source DNS domain (`app1.apps.ocp3.example.com`), you must create a route in the application’s project for the host name that was used in the source cluster:
```

Comment:
Creating a route and a DNS record apply to BOTH http and https. The current structure suggests that those steps are only for http.